lib/engines/shopping_cart/lib/concerns/controllers/application_controller.rb

```diff
@@ -0,0 +1,22 @@
+module ShoppingCart
+  module Concerns
+    module Controllers
+      module ApplicationController
```

Comment:
@b-artem IMO this module should have another name, it's strange when you include it in `ApplicationController` in main app then you inherit engine's `ApplicationController` from main app `ApplicationController` and also include this module. 

Do you really need this module in both controllers or it's mistake?

Reply:
@Dulenchik Thank you, that was a mistake. Corrected it. This concern is included in main app `ApplicationController` only now. Renamed it to `CartSupport` also.